src/server/scripts/Spells/spell_dk.cpp

```diff
@@ -446,7 +446,7 @@ class spell_dk_death_pact : public AuraScript
     void HandleCalcAmount(AuraEffect const* /*aurEff*/, int32& amount, bool& /*canBeRecalculated*/)
     {
         if (Unit* caster = GetCaster())
-            amount = int32(caster->CountPctFromMaxHealth(amount));
+            amount = int32(caster->CountPctFromMaxHealth(GetEffectInfo(EFFECT_2).CalcValue(caster)));
```

Comment:
does this need a Validate check for EFFECT_2 now or is that checked already somewhere ?

Reply:
it does need it

Reply:
done